clang/lib/Sema/SemaExprMember.cpp

```diff
@@ -1055,9 +1045,19 @@ Sema::BuildMemberReferenceExpr(Expr *BaseExpr, QualType BaseExprType,
       }
     }
 
-    Diag(R.getNameLoc(), diag::err_no_member)
-      << MemberName << DC
-      << (BaseExpr ? BaseExpr->getSourceRange() : SourceRange());
+    if (SS.isNotEmpty() && !DC) {
+      Diag(R.getNameLoc(), diag::err_undeclared_use)
+          << MemberName << SS.getRange();
+    } else if (DC) {
+      Diag(R.getNameLoc(), diag::err_no_member)
+          << MemberName << DC
+          << (BaseExpr ? BaseExpr->getSourceRange() : SourceRange());
+    } else {
+      // FIXME: Is this needed?
```

Comment:
Did you try just asserting here?  Also: I think I'd prefer us just merging this and the above branch, potentially with a ternary on `DC`.

Reply:
Still this.

Reply:
Yup, I will be addressing this shortly